### ISSUE TITLE
[C#] Implement new Concept Exercise: explicit casts

### ISSUE THREAD

**Issue by mikedamay** [exercism/v3#1650]
This issue describes how to implement the `explicit-cast` concept exercise for the C# track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

The goal of this exercise is to teach the student how to use `time` in C#.

## Learning objectives

- Know how to down cast in C#.
- Know how to cast safely.
- Understand that implicit casting is ubiquitous.
- Know the difference between down casts using `as` and those using the old style C syntax.

## Out of scope

- Casting of primitive types (discussed under signed-integers).
- Pattern matching (discussed under `pattern-matching-types`).

## Concepts

- `explicit-casts` with `as` keyword.
-  `implicit-casts`

## Prerequisites

- `classes`
- `inheritance`

Other prerequisites will depend on the story.

### Hints
This [article][as-operator] describes how to use the `as` operator to down cast a reference type.
This [article][is-operator] describes how to use the `is` operator as a guard when casting

### After
This [article][as-operator] describes how to use the `as` operator to down cast a reference type.
This [article][is-operator] describes how to use the `is` operator as a guard when casting
This [article][cast-expression] describes use of the old C style cast expression.

## Representer

This exercise does not require any specific representation logic to be added to the [representer][representer].

## Analyzer

A check that the `is` guard is applied as appropriate can be checked by the [analyzer][analyzer].

## Implementing

To implement this exercise, please [follow these instructions](https://github.com/exercism/v3/blob/master/languages/csharp/reference/implementing-a-concept-exercise.md).

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[as-operator]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#as-operator
[is-operator]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#the-is-operator
[cast-expression]: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/type-testing-and-cast#cast-expression
[analyzer]: https://github.com/exercism/csharp-analyzer
[representer]: https://github.com/exercism/csharp-representer

**Comment by mikedamay** [comment]
Duplicate of #1142.
